chat-plugins/dexsearch.js

```diff
@@ -11,13 +11,156 @@
 
 const RESULTS_MAX_LENGTH = 10;
 
-exports.commands = {
-	ds: 'dexsearch',
-	dsearch: 'dexsearch',
-	dexsearch: function (target, room, user, connection, cmd, message) {
-		if (!this.canBroadcast()) return;
+if (!process.send) {
+	// This is the parent
+	let child = exports.child = require('child_process').fork('dexsearch.js', {cwd: __dirname});
 
-		if (!target) return this.parse('/help dexsearch');
+	let resolvers = {};
+	let makePromise = function (id) {
+		return new Promise((resolve, reject) => {
+			resolvers[id] = resolve;
+		});
+	};
```

Comment:
`makePromise` is the wrong abstraction level. Your function should handle both the ID generation and `child.send`.

Also when I said to make `id = Math.random()`, that was just an example. Actual IDs are generally counters; see how `guid` works in `verifier.js`.